### PR TITLE
Add list data to destructured types

### PR DESCRIPTION
**Pull request by minasmart**
transforms fields on the type data from something like

```javascript
fields: [
  'title',
  'handle',
  'images'
]
```

to:

```javascript
fields: {
  title: { isList: true },
  handle: { isList: true },
  images: { isList: true }
}
```

### CODE REVIEW

**Comment by minasmart** [comment]
@lreeves and @swalkinshaw could you take a quick look?

File: graph/generator.js [hunk 1]
```diff
@@ -5,31 +5,42 @@ import path from 'path';
 import dasherize from './dasherize';
 import isBuiltin from './is-builtin';
 import getBaseType from './get-base-type';
+import hasListType from './has-list-type';
```
Comment (swalkinshaw) [inline graph/generator.js]: This file doesn't exist?

Comment (minasmart) [inline graph/generator.js]: I didn't git add because I'm a dumb. Please hold.

File: graph/has-list-type.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export default function hasListType(typeDescriptor) {
+  if (typeDescriptor.kind === 'LIST') {
+    return true;
+  } else if (typeDescriptor.ofType) {
+    return hasListType(typeDescriptor.ofType);
+  }
+
+  return false;
+}
```
Comment (minasmart) [inline graph/has-list-type.js]: @swalkinshaw

**Comment by swalkinshaw** [comment]
👍

**Pull request merged**